### DISCUSSION TITLE
Working with Rebar3

### DISCUSSION THREAD

**Discussion by yenrabbyui** [ninenines/cowboy#1567] [Q&A] [answered]
I'm trying to get the ssl example running using rebar3 and have an issue popping up. I've moved the src and priv directories into my rebar3 app called "junk." I've also updated the dependencies to cowboy 2.9.0. Any ideas would be appreciated as I'm stuck. 

There error is
`exited: {bad_return,
             {{junk_app,start,[normal,[]]},
              {'EXIT',
               {noproc,
                {gen_server,call,
                 [ranch_sup,
                  {start_child,
                   {{ranch_listener_sup,https},
                    {ranch_listener_sup,start_link,
                     [https,ranch_ssl,
                      #{connection_type => supervisor,
                        socket_opts =>
                         [{next_protocols_advertised,
                           [<<"h2">>,<<"http/1.1">>]},
                          {alpn_preferred_protocols,[<<"h2">>,<<"http/1.1">>]},
                          {port,443},
                          {cacertfile,
                           "/home/pseudo/junk/_build/default/lib/junk/priv/ssl/cowboy-ca.crt"},
                          {certfile,
                           "/home/pseudo/junk/_build/default/lib/junk/priv/ssl/server.crt"},
                          {keyfile,
                           "/home/pseudo/junk/_build/default/lib/junk/priv/ssl/server.key"}]},
                      cowboy_tls,
                      #{connection_type => supervisor,
                        env =>
                         #{dispatch => [{'_',[],[{[],[],toppage_h,[]}]}]}}]},
                    permanent,infinity,supervisor,
                    [ranch_listener_sup]}},
                  infinity]}}}}}
    type: temporary`

**Comment by essen** [comment, marked as answer]
Well it seems you haven't started Cowboy. With Rebar3 you will probably need to add cowboy to the .app.src file.
